homeassistant/components/media_player/yamaha.py

```diff
@@ -0,0 +1,127 @@
+"""
+import logging
+import rxv
+
+from homeassistant.components.media_player import (
+    DOMAIN, SUPPORT_NEXT_TRACK, SUPPORT_PAUSE, SUPPORT_PREVIOUS_TRACK,
+    SUPPORT_TURN_OFF, SUPPORT_TURN_ON, SUPPORT_VOLUME_MUTE, SUPPORT_VOLUME_SET,
+    MediaPlayerDevice)
+from homeassistant.const import CONF_HOST, STATE_OFF, STATE_ON, STATE_UNKNOWN
+REQUIREMENTS = ['rxv==0.1.9']
+_LOGGER = logging.getLogger(__name__)
+_GLOBALRECEIVER = None
+
+SUPPORT_YAMAHA = SUPPORT_VOLUME_SET | SUPPORT_VOLUME_MUTE | \
+    SUPPORT_TURN_ON | SUPPORT_TURN_OFF
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the Denon platform."""
+    yamaha = YamahaDevice()
+    if yamaha.update():
+        add_devices([yamaha])
+        return True
+    else:
+        return False
+
+
+class YamahaDevice(MediaPlayerDevice):
+    """Representation of a Yamaha device."""
+
+    # pylint: disable=too-many-public-methods, abstract-method
+    def __init__(self):
+        """Initialize the Yamaha Receiver"""
+        self._name = "Yamaha Receiver"
+        self._host = ""
+        self._pwstate = "Off"
+        self._muted = False
+        self._mediasource = ""
+
+    def update(self):
+        """Get the latest details from the device."""
+        try:
+            receivers = rxv.find()
+            global _GLOBALRECEIVER
+            _GLOBALRECEIVER= receivers[0]
+        except ConnectionRefusedError:
+            return False
+        
+        if _GLOBALRECEIVER.on == True:
+            self._pwstate = STATE_ON
+        if _GLOBALRECEIVER.on == False:
+            self._pwstate = STATE_OFF
+
+        self._volume = .5
+        self._muted = False
+        self._mediasource = _GLOBALRECEIVER.input
+
+        return True
+
+    @property
+    def name(self):
+        """Return the name of the device."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the state of the device."""
+        if self._pwstate == STATE_OFF:
+            return STATE_OFF
+        if self._pwstate == STATE_ON:
+            return STATE_ON
+
+        return STATE_UNKNOWN
+
+    @property
+    def volume_level(self):
+        """Volume level of the media player (0..1)."""
+        return self._volume
+
+    @property
+    def is_volume_muted(self):
+        """Boolean if volume is currently muted."""
+        return self._muted
+
+    @property
+    def media_title(self):
+        """Current media source."""
+        return self._mediasource
+
+    @property
+    def supported_media_commands(self):
+        """Flag of media commands that are supported."""
+        return SUPPORT_YAMAHA
+
+    def turn_off(self):
+        """Turn off media player."""
+        _GLOBALRECEIVER.on = False
+
+    def volume_up(self):
+        """Volume up media player."""
+        _GLOBALRECEIVER.volume = (self._volume)
+
+    def volume_down(self):
+        """Volume down media player."""
+        _GLOBALRECEIVER.volume = (self._volume - 1)
```

Comment:
self._volume is 0.5 and you're substracting 1. Which scale do you use? Home Assistant uses 0…1